### DISCUSSION TITLE
Issue installing pdfplumber on M1 Mac

### DISCUSSION THREAD

**Discussion by Anlanther** [jsvine/pdfplumber#817] [Q&A] [answered]
I was previously able to import pdfplumber no problem one month ago on the same computer I am using now, however I am now having issues importing. I have tried reinstalling pdfplumber from a normal `pip install pdfplumber` and I've already made sure that VSC and all of my VSC extensions are up to date.

Says here that I have incompatible architecture. Just wanted to make sure incompatible architecture is the sole reason to why this is not working, and if it will be fixed? Especially since I was able to use this library previously with no issues with the same machine.

Here is my error:
```py
ImportError                               Traceback (most recent call last)
/var/folders/2f/lhp05thj49g1v2h0_gwdjyyh0000gn/T/ipykernel_74676/1777862011.py in <module>
----> 1 import pdfplumber
      2 import pandas as pd
      3 import numpy as np
      4 from datetime import datetime
      5 import os

/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pdfplumber/__init__.py in <module>
      8 
      9 import pdfminer
---> 10 import pdfminer.pdftypes
     11 
     12 from . import utils

/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pdfminer/pdftypes.py in <module>
     20 from .ccitt import ccittfaxdecode
     21 from .lzw import lzwdecode
---> 22 from .psparser import LIT
     23 from .psparser import PSException
     24 from .psparser import PSObject

/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pdfminer/psparser.py in <module>
     20 
...
----> 9 from .md import is_suspiciously_successive_range
     10 from .models import CoherenceMatches
     11 from .utils import (

ImportError: dlopen(/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/charset_normalizer/md.cpython-310-darwin.so, 0x0002): tried: '/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/charset_normalizer/md.cpython-310-darwin.so' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64'))
```

```sh
where pdfplumber
/Users/xyz/opt/anaconda3/bin/pdfplumber

where python
python: aliased to python3
/Users/xyz/opt/anaconda3/bin/python
```

**Comment by jsvine** [comment, marked as answer]
Hi @Anlanther, and thanks for you interest in this library. Looking at the error message (and thank you for providing that), it appears the the error is stemming from the `charset_normalizer` package, imported by `pdfminer.six` (one of `pdfplumber`'s main dependencies).

It's possible that this has something to do with how `anaconda` is handling the `charset_normalizer` package, but I don't use `anaconda` so I couldn't say for sure.

**Reply by Anlanther** [reply]
Changed my Python environment and it works now, thanks
